be/src/runtime/timestamp_value.h

```diff
@@ -0,0 +1,174 @@
+        val = timestamp;
+    }
+
+    bool from_date_time_value(const DateTimeValue& tv, std::string timezone) {
+        boost::local_time::time_zone_ptr local_time_zone = TimezoneDatabase::find_timezone(timezone);
+        if (local_time_zone == nullptr) {
+            return false;
+        }
+
+        std::stringstream ss;
+        ss << tv;
+        boost::posix_time::ptime pt = boost::posix_time::time_from_string(ss.str());
+        boost::local_time::local_date_time lt(pt.date(), pt.time_of_day(), local_time_zone,
+                                              boost::local_time::local_date_time::NOT_DATE_TIME_ON_ERROR);
+
+        boost::posix_time::ptime utc_ptime = lt.utc_time();
+        boost::posix_time::ptime utc_start(boost::gregorian::date(1970, 1, 1));
+        boost::posix_time::time_duration dur = utc_ptime - utc_start;
+        val = dur.total_milliseconds();
+        return true;
+    }
+
+    bool to_datetime_value(DateTimeValue& dt_val, std::string timezone) {
+        boost::local_time::time_zone_ptr local_time_zone = TimezoneDatabase::find_timezone(timezone);
+        if (local_time_zone == nullptr) {
+            return false;
+        }
+        boost::local_time::local_date_time lt(boost::posix_time::from_time_t(val), local_time_zone);
+        boost::posix_time::ptime locat_ptime = lt.local_time();
+
+        dt_val.set_type(TIME_DATETIME);
+        dt_val.from_olap_datetime(
+                locat_ptime.date().year() * 10000000000 +
+                locat_ptime.date().month() * 100000000 +
+                locat_ptime.date().day() * 1000000 +
+                locat_ptime.time_of_day().hours() * 10000 +
+                locat_ptime.time_of_day().minutes() * 100 +
+                locat_ptime.time_of_day().seconds());
+        return true;
+    }
+
+    std::string to_datetime_string(std::string timezone) {
+        boost::local_time::time_zone_ptr local_time_zone = TimezoneDatabase::find_timezone(timezone);
+        if (local_time_zone == nullptr) {
+            return "";
+        }
+        boost::local_time::local_date_time lt(boost::posix_time::from_time_t(val), local_time_zone);
+        boost::posix_time::ptime ret_ptime = lt.local_time();
+
+        std::stringstream ss;
+        ss << std::setw(4) << std::setfill('0') << ret_ptime.date().year() << "-"
+           << std::setw(2) << std::setfill('0') << ret_ptime.date().month().as_number() << "-"
+           << std::setw(2) << std::setfill('0') << ret_ptime.date().day() << " "
+           << std::setw(2) << std::setfill('0') << ret_ptime.time_of_day().hours() << ":"
+           << std::setw(2) << std::setfill('0') << ret_ptime.time_of_day().minutes() << ":"
+           << std::setw(2) << std::setfill('0') << ret_ptime.time_of_day().seconds();
+        return ss.str();
+    }
+
+    void to_datetime_val(doris_udf::DateTimeVal *tv) const {
+        boost::posix_time::ptime p = boost::posix_time::from_time_t(val / 1000);
+        int _year = p.date().year();
```

Comment:
Your code style is not compatible with Doris C++ code style.
private member should starts with `_`.
local variables should not start with `_`